components/form/demo/inline-login.tsx

```diff
@@ -4,11 +4,11 @@ import { Button, Form, Input } from 'antd';
 
 const App: React.FC = () => {
   const [form] = Form.useForm();
-  const [, forceUpdate] = useState({});
+  const [btnDisabled, setBtnDisabled] = useState<boolean>(true);
 
   // To disable submit button at the beginning.
   useEffect(() => {
-    forceUpdate({});
+    setBtnDisabled(false);
```

Comment:
```suggestion
    setBtnDisabled(btnDisabled ||
              !form.isFieldsTouched(true) ||
              !!form.getFieldsError().filter(({ errors }) => errors.length).length);
```
把条件都放进来？不然只是做了一个客户端渲染的判断

Reply:
判断条件必须放在 jsx 的函数里面，放在外面不会触发更新